### PR TITLE
RavenDB-14049 - Map-Reduce isn't updated after documents deletion

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-14051
https://issues.hibernatingrhinos.com/issue/RavenDB-14058

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 1]
```diff
@@ -114,6 +114,9 @@ public abstract class Index : ITombstoneAware, IDisposable, ILowMemoryHandler
         private readonly ConcurrentDictionary<string, IndexProgress.CollectionStats> _inMemoryIndexProgress =
             new ConcurrentDictionary<string, IndexProgress.CollectionStats>();
 
+        private readonly ConcurrentDictionary<string, IndexProgress.CollectionStats> _inMemoryReferencesIndexProgress =
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Indexes/Index.cs:117]: Shouldn't we use `StringComparer.OridinalIgnoreCase` here (and in the other stats) ?

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Documents/Indexes/Index.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Index.cs:117]: Nvm, this comes from compiled.ReferencedCollections which has that compiler.

Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Index.cs:117]: Comparer*

**Comment by arekpalinski** [comment]
LGTM

**Pull request merged**